Sources/FoundationEssentials/Subprocess/Subprocess+Configuration.swift

```diff
@@ -0,0 +1,615 @@
+        }
+
+        public init(executablePathOverride: String?, remainingValues: [String]) {
+            self.storage = remainingValues.map { .string($0) }
+            if let executablePathOverride = executablePathOverride {
+                self.executablePathOverride = .string(executablePathOverride)
+            } else {
+                self.executablePathOverride = nil
+            }
+        }
+
+        public init(executablePathOverride: Data?, remainingValues: [Data]) {
+            self.storage = remainingValues.map { .rawBytes($0.toArray()) }
+            if let override = executablePathOverride {
+                self.executablePathOverride = .rawBytes(override.toArray())
+            } else {
+                self.executablePathOverride = nil
+            }
+        }
+    }
+}
+
+// MARK: - Environment
+extension Subprocess {
+    public struct Environment: Sendable {
+        internal enum Configuration {
+            case inherit([StringOrRawBytes : StringOrRawBytes])
+            case custom([StringOrRawBytes : StringOrRawBytes])
+        }
+
+        internal let config: Configuration
+
+        init(config: Configuration) {
+            self.config = config
+        }
+
+        public static var inherit: Self {
+            return .init(config: .inherit([:]))
+        }
+
+        public func updating(_ newValue: [String : String]) -> Self {
+            return .init(config: .inherit(newValue.wrapToStringOrRawBytes()))
+        }
+
+        public func updating(_ newValue: [Data : Data]) -> Self {
+            return .init(config: .inherit(newValue.wrapToStringOrRawBytes()))
+        }
+
+        public static func custom(_ newValue: [String : String]) -> Self {
+            return .init(config: .custom(newValue.wrapToStringOrRawBytes()))
+        }
+
+        public static func custom(_ newValue: [Data : Data]) -> Self {
+            return .init(config: .custom(newValue.wrapToStringOrRawBytes()))
+        }
+    }
+}
+
+fileprivate extension Dictionary where Key == String, Value == String {
+    func wrapToStringOrRawBytes() -> [Subprocess.StringOrRawBytes : Subprocess.StringOrRawBytes] {
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.string(key)] = .string(value)
+        }
+        return result
+    }
+}
+
+fileprivate extension Dictionary where Key == Data, Value == Data {
+    func wrapToStringOrRawBytes() -> [Subprocess.StringOrRawBytes : Subprocess.StringOrRawBytes] {
+        var result = Dictionary<
+            Subprocess.StringOrRawBytes,
+            Subprocess.StringOrRawBytes
+        >(minimumCapacity: self.count)
+        for (key, value) in self {
+            result[.rawBytes(key.toArray())] = .rawBytes(value.toArray())
+        }
+        return result
+    }
+}
+
+fileprivate extension Data {
+    func toArray<T>() -> [T] {
```

Comment:
Could you elaborate a bit more on why this is necessary? I would have naively assumed that anything that we need to do with the `Data` shouldn't require copying the data out to an `Array`